### DISCUSSION TITLE
Porting some Lua packages to Teal

### DISCUSSION THREAD

**Discussion by catwell** [teal-language/tl#887] [Show and tell]
I wrote a [small gfshare implementation](https://github.com/catwell/cw-lua/tree/master/gfshare) in Lua for fun and converted it to Teal.

Its tests used [base2base](https://github.com/oploadk/base2base) (a base-to-base converter, here to convert to and from hex). I started by writing a .d.tl file but eventually I converted base2base to Teal (it was easy).

... but the base2base tests used [cwtest](https://github.com/catwell/cwtest) (a test helper), so eventually I converted it to Teal as well (it was... a bit less easy, but still manageable :p). I published both cwtest and base2base to luarocks.

I will probably keep doing this with more of my Lua packages, Teal is now usable enough. I just wish [luarocks had better support for Teal](https://github.com/teal-language/tl/discussions/836) (in particular I wish I could use `luarocks test` like I did before) but this is hardly enough to keep me from using Teal by default now.

**Comment by Aire-One** [comment]
> I will probably keep doing this with more of my Lua packages, Teal is now usable enough. I just wish https://github.com/teal-language/tl/discussions/836 (in particular I wish I could use luarocks test like I did before) but this is hardly enough to keep me from using Teal by default now.

The particular case of `luarocks test` is also discussed in https://github.com/teal-language/tl/discussions/872#discussioncomment-11530442 (and replies), at least in the scope of using Busted with Teal.
